pkg/binding/bindingpolicy.go

```diff
@@ -113,16 +129,33 @@ func (c *Controller) requeueForBindingPolicyChanges() error {
 	return c.requeueWorkloadObjects()
 }
 
-func (c *Controller) getBindingPolicyByName(name string) (runtime.Object, error) {
-	lister := c.listers["control.kubestellar.io/v1alpha1/BindingPolicy"]
-	if lister == nil {
-		return nil, fmt.Errorf("could not get lister for placememt")
+func (c *Controller) requeueSelectedWorkloadObjects(bindingPolicyName string) error {
+	if !c.bindingPolicyResolver.ResolutionExists(bindingPolicyName) {
+		return nil
 	}
-	got, err := lister.Get(name)
+
+	// requeue all objects that are selected by the bindingpolicy (are in its resolution)
+	objectKeys, err := c.bindingPolicyResolver.GetObjectKeys(bindingPolicyName)
 	if err != nil {
-		return nil, err
+		return fmt.Errorf("failed to get object keys from bindingpolicy resolver for bindingpolicy %s: %w",
+			bindingPolicyName, err)
+	}
+
+	objs := make([]runtime.Object, 0, len(objectKeys))
+	for _, key := range objectKeys {
+		obj, err := c.getObjectFromKey(*key)
+		if err != nil {
+			return fmt.Errorf("failed to get object from key: %w", err)
+		}
+
+		objs = append(objs, obj)
```

Comment:
What if `obj == nil` (which c.getObjectFromKey can return with a `nil` error)?

Reply:
Right I missed that. Do we want to keep getObjectFromKey that way? for now I added an `|| obj == nil` to the check.

Reply:
It seems a little treacherous to me the way it is now, as demonstrated by the fact that you missed this. Does not have to be addressed in this PR.

Reply:
Opened issue #1790